src/datasources/email/email.datasource.spec.ts

```diff
@@ -0,0 +1,271 @@
+import { EmailDatasource } from '@/datasources/email/email.datasource';
+import * as postgres from 'postgres';
+import { PostgresError } from 'postgres';
+import { faker } from '@faker-js/faker';
+import { Email } from '@/datasources/email/entities/email.entity';
+import { VerificationStatus } from '@/datasources/email/entities/verification.status';
+import { EmailAlreadyVerifiedError } from '@/datasources/email/errors/email-already-verified.error';
+import { EmailAddressDoesNotExistError } from '@/datasources/email/errors/email-address-does-not-exist.error';
+import { InvalidVerificationCodeError } from '@/datasources/email/errors/invalid-verification-code.error';
+import { FakeConfigurationService } from '@/config/__tests__/fake.configuration.service';
+import { ExpiredVerificationCodeError } from '@/datasources/email/errors/expired-verification-code.error';
+import * as shift from 'postgres-shift';
+
+describe('Email Datasource Tests', () => {
+  let target: EmailDatasource;
+
+  const sql = postgres({
+    host: process.env.POSTGRES_TEST_HOST || 'localhost',
+    port: parseInt(process.env.POSTGRES_TEST_PORT || `${5433}`),
+    db: process.env.POSTGRES_TEST_DB || 'test-db',
+    user: process.env.POSTGRES_TEST_USER || 'postgres',
+    password: process.env.POSTGRES_TEST_PASSWORD || 'postgres',
+  });
```

Comment:
What about providing this configuration via `__tests__/configuration.ts`? Randomly generated values for `db.postgres` on there won't be used. Plus, moving this configuration there would make it reusable by other database-related test files. No strong opinion on this, just suggesting.

Reply:
It should be ok to move it there I agree. However we would need to call `configuration()` either way on this test since the `ConfigurationModule` is not part of this test.

Reply:
Ok, no strong opinion regarding this. We can consider it in the future if we find ourselves copypasting this code in another test class 👍🏻

Reply:
I think it makes sense to tackle it now. Changed here [466eea5](https://github.com/safe-global/safe-client-gateway/pull/829/commits/466eea5dfa0ede5e66a16337a30739e31d0ab6f9).

Thanks for the suggestion! 🙌